src/Mgate/PersonneBundle/Entity/Membre.php

```diff
@@ -164,6 +169,37 @@ public function __toString()
         return $this->getPersonne()->__toString();
     }
 
+    public function anonymize()
+    {
+        $this->dateConventionEleve = null;
+        $this->identifiant = null;
+        $this->emailEMSE = null;
+        $this->promotion = null;
+        $this->dateDeNaissance = null;
+        $this->lieuDeNaissance = null;
+        $this->nationalite = null;
+        $this->photoURI = null;
+        $this->formatPaiement = null;
+        $this->securiteSociale = null;
+        $this->commentaire = null;
+
+        /** remove non critical (business related) relations */
+        /** @var Competence $c */
```

Comment:
You don't need this (the phpdoc block) here. If you want an IDE to pick its type you should instead do that in the property declaration:

```php
/**
 * @var Competence[]|Collection
 * @ORM...
 *
private $competences;
```

Same for the others below. I also recommend to use `$competence` instead of `$c`, there is no good reason for those short variable names :)